### PR TITLE
Add a Shimmer effect

### LINKED ISSUE #33: I have a prototype for a Shimmer effect.
I have a naive prototype for a Shimmer effect (a la [SwiftUI-Shimmer](https://github.com/markiv/SwiftUI-Shimmer) ) that I would like to add to the library. 

I was able to get the shine working, but the animatable value isn't working correctly. It's pivoting at the top leading corner, and I'd like it to simply slide horizontally. If anyone, has any pointers, that would be much appreciated. 

I tried to use the same "uv" method that Paul showed [here](https://youtu.be/EgzWwgRpUuw?feature=shared&t=1749). 

https://github.com/twostraws/Inferno/assets/72824209/71580d43-dd73-4450-8aad-4fb756f44237

### PR DESCRIPTION
**Pull request by bwhtmn**
Add an animated shimmer effect.
Resolves #33.

https://github.com/twostraws/Inferno/assets/1206592/c46615e2-5012-4574-beb1-7307c647738c

Created the shimmering by modifying lightness rather than alpha as in [SwiftUI-Shimmer](https://github.com/markiv/SwiftUI-Shimmer) that was linked to in the issue. This lightness approach was used since I found that non-opaque alpha values applied in a shader don't show the background views, probably because the shader is applied in the final rendering pass.

### CODE REVIEW

File: Sources/Inferno/Shaders/Transformation/Shimmer.metal [hunk 1]
```diff
@@ -0,0 +1,119 @@
+//
+//  Shimmer.metal
+//  Inferno
+//
+//  Created by Brent Whitman on 2024-04-10.
```
Comment (twostraws) [inline Sources/Inferno/Shaders/Transformation/Shimmer.metal:5]: Please copy in the same header comments as the other files; it directs folks to the repository and license.

File: README.md [hunk 2]
```diff
@@ -1601,6 +1653,7 @@ Some shaders were ported to Metal by me, from other open-source samples also rel
 - Circle, Circle Wave, Diamond, Diamond Wave are based on [Polka Dots Curtain](https://gl-transitions.com/editor/PolkaDotsCurtain) by bobylito.
 - Crosswarp is based on [Crosswarp](https://gl-transitions.com/editor/crosswarp) by Eke Péter.
 - Radial is based on [Radial](https://gl-transitions.com/editor/Radial) by Xaychru / gre.
+- Shimmer is inspired by [SwiftUI-Shimmer](https://github.com/markiv/SwiftUI-Shimmer) by markiv.
```
Comment (twostraws) [inline README.md:1656]: I would add your name here too; this is the correct place for credits.

**Comment by bwhtmn** [comment]
> Great job – thank you! I've added two small comments, but otherwise this can be merged.

Thanks for the quick review! 😄 I've addressed both of your comments.

**Comment by twostraws** [comment]
Perfect! 🙌

**Pull request merged**